### PR TITLE
Collect user data for Legal module

### PR DESCRIPTION
**Pull request by yurabakhtin**
https://github.com/humhub/humhub-internal/issues/167

### CODE REVIEW

File: Events.php [hunk 1]
```diff
@@ -114,6 +115,20 @@ public static function onRestApiAddRules()
         ], 'wiki');
     }
 
+    public static function onLegalExportServiceCollectUserData($event)
+    {
+        if (!$event instanceof \humhub\modules\legal\events\UserDataCollectionEvent) {
```
Comment (luke-) [inline Events.php:120]: @yurabakhtin Do we need this check here?

**Review by luke- — CHANGES_REQUESTED** [review]

**Review by luke- — CHANGES_REQUESTED** [review]
@yurabakhtin One small note. 

Then we could add this code as reference into the legal module `DEVELOPER.md`

File: Events.php (hunk 1 above)
Comment (yurabakhtin) [reply Events.php:120]: @luke- We can remove that if module config will be correct for all other modules as `['class' => 'humhub\modules\legal\services\ExportService', 'event' => 'collectUserData',`.
Or we can replace it with `/* @var \humhub\modules\legal\events\UserDataCollectionEvent $event */` in order to keep a declaration for the method `addUserData` in PhpStormeditor.
Or `public static function onLegalExportServiceCollectUserData(\humhub\modules\legal\events\UserDataCollectionEvent $event)`, but in such case I think we may have an error if the module Legal is not installed.

Comment (luke-) [reply Events.php:120]: @yurabakhtin On my tests there were not error on unknown symbols when there are not used. Then I would go with following style/naming.

```php
use \humhub\modules\legal\events\UserDataCollectionEvent;

class Events {

    public static function onLegalModuleUserDataExport(UserDataCollectionEvent $event)
    {
         //...
    }

}

```

Comment (yurabakhtin) [reply Events.php]: @luke- Yes, you are right, it works wirhout errors, done in the commit https://github.com/humhub/wiki/pull/348/commits/922b0a01bb4452e54d0db6192793fc402d4054fd.

**Review by luke- — APPROVED** [review]

**Pull request merged**